### ISSUE TITLE
CNN + LSTM Input shape parameters doubt

### ISSUE THREAD

**Issue by dbsousa01** [keras-team/keras#10413]
Hello all! 

I am trying to train a CNN (VGG-Face if wondering) and afterwards put a LSTM layer to do an analysis overtime. My doubt is how do I organize my input dataset. I have for example 10 videos and in each video we have 50 frames each frame is 224x224 with 3 channels RGB so I suppose the input of my network should be (10,50,224,224,3) where 50 is the time-step for the LSTM. Now my question is load every frame to a numpy array where I have (500,224,224,3) (10x50 = 500). How do I reshape this vector to be the desired one? I have tried a simple reshape but it seems it is not working. Am I doing something wrong? 

How can I associate each frame(1 to 50) belonging to a specific video (1 to 10) and how can the LSTM understand that?

**Comment by SebiSebi** [comment]
Hi,

An LSTM cell expects a 3D input tensor with shape `(batch_size, timesteps, input_dim)`. You can use a [Reshape](https://keras.io/layers/core/#reshape) layer like so: `model.add(Reshape((50, 224 * 224 * 3)))`. before the LSTM. Note that the `Reshape` layer does not include the batch axis (in your case it seems to be set to 10). However, I would not recommend you to use an LSTM like so because:

- It is going to have a lot of params since `224 * 224 * 3 = 150528` (which is huge);
- If you have only 10 videos, it is very likely that you will not succeed in training the LSTM.

**Comment by dbsousa01** [comment]
Thanks @SebiSebi ! I found that I had two possible solutions:

- Reshape Layer: But as @SebiSebi said it takes a huuge amount of memory;

- Flatten Layer: Also works, might be the best solution

Since I was doing video texture analysis. I have chosen the new layer supported in keras ConvLSTM2D where it joins a normal conv layer with lstm and this way I joined a CNN + LSTM
